mi-osu-api/src/auth.rs

```diff
@@ -87,13 +87,15 @@ pub struct AuthResponseBody {
 async fn request_token(
     client: &Client,
     body: AuthRequest,
-) -> Result<AuthResponseBody, ReqwestError> {
+) -> Result<AuthResponseBody, OsuApiError> {
     let response_result = client
         .post("https://osu.ppy.sh/oauth/token")
         .form(&body)
         .send()
         .await?;
-    response_result.json::<AuthResponseBody>().await
+    let response_result: Result<AuthResponseBody, OsuApiError> =
+        response_result.try_deserialising().await;
+    response_result
```

Comment:
```suggestion
    response_result.try_deserialising().await
```
Can you change this for all alike calls?